Install xdebug in php80-fpm

janmartenjongerius opened the pull request:
Install the xdebug extension in the PHP 8 FPM image.

The corresponding xdebug settings have been upgraded. The configuration
`xdebug.discover_client_host` has been set to `true` to allow flexible
host configuration for IDEs through HTTP headers.

/cc @JeroenBoersma

Commit:
Install xdebug in php80-fpm

Install the xdebug extension in the PHP 8 FPM image.

The corresponding xdebug settings have been upgraded. The configuration
`xdebug.discover_client_host` has been set to `true` to allow flexible
host configuration for IDEs through HTTP headers.

janmartenjongerius commented on php80/fpm/conf/php.ini:
```diff
@@ -5,7 +5,7 @@ memory_limit = 512M
-xdebug.remote_enable = 1
-xdebug.remote_host = 172.17.0.1
+xdebug.client_host = 172.17.0.1
+xdebug.discover_client_host = true
```
See https://xdebug.org/docs/all_settings#discover_client_host

janmartenjongerius commented on php80/fpm/conf/php.ini:
```diff
@@ -5,7 +5,7 @@ memory_limit = 512M
 log_errors = 1
 error_reporting = E_ALL
 
-xdebug.remote_enable = 1
```
This setting has been replaced by `xdebug.mode` with value `debug`.
- https://xdebug.org/docs/upgrade_guide

JeroenBoersma commented on php80/fpm/Dockerfile:
```diff
@@ -35,6 +35,11 @@ RUN docker-php-ext-install bcmath mysqli pdo_mysql soap zip intl opcache xsl pcn
+# Install xdebug
+RUN pecl install xdebug \
+    && docker-php-ext-enable xdebug \
+    && rm -rf /tmp/pear
```
pecl was removed from php 8, was t re-added?

janmartenjongerius replied:
When I run a `docker build` and then a `docker run` on the built image with `php -v`, it shows xdebug installed.

![image](https://user-images.githubusercontent.com/642417/113691655-f4d49780-96cc-11eb-80fe-5806abc9fb51.png)

Leave it to the author of xdebug to fix this for us: docker-library/php#1087

janmartenjongerius replied:
So, officially it wasn't removed, just not enabled. It explicitly required `--with-pear` it seems, which has been added to the official PHP images.

johmanx10 commented:
When I run a docker build and then a docker run on the built image with php
-v, it shows xdebug installed.

Leave it to the author of xdebug to fix this for us:
https://github.com/docker-library/php/issues/1087

On Tue, Apr 6, 2021, 08:40 Jeroen Boersma ***@***.***> wrote:

> *@JeroenBoersma* requested changes on this pull request.
> ------------------------------
>
> In php80/fpm/Dockerfile
> <https://github.com/JeroenBoersma/docker-php/pull/10#discussion_r607561492>
> :
>
> > @@ -35,6 +35,11 @@ RUN docker-php-ext-install bcmath mysqli pdo_mysql soap zip intl opcache xsl pcn
>      && docker-php-ext-configure gd --with-freetype --with-jpeg --with-webp \
>      && docker-php-ext-install gd
>
> +# Install xdebug
> +RUN pecl install xdebug \
> +    && docker-php-ext-enable xdebug \
> +    && rm -rf /tmp/pear
>
> pecl was removed from php 8, was t re-added?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JeroenBoersma/docker-php/pull/10#pullrequestreview-628619235>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAE424OX2KTYVX3STKSZSELTHKUGXANCNFSM42MURDMQ>
> .
>

janmartenjongerius commented:
It dawns on me that earlier versions also use xdebug 3. I had a look and my proposed settings deviate from what's already there.

Differences are:

- The setting `xdebug.discover_client_host` is added
- The setting `xdebug.mode` is set to `coverage,debug,develop` instead of `develop,debug` in other versions.
  > Adding `coverage` allows things like running PHPUnit with code coverage enabled.

Would you prefer I either:

- Make PHP 8 config the same as other version
- Update the config of other versions to be like my proposal gor PHP 8

Basically, the question is whether you agree with the changes I propose.

/cc @JeroenBoersma

Commit:
Make xdebug 3 settings consistent

PHP versions ^7.2 support xdebug version 3.
Their configurations have been updated to follow the configuration
changes for PHP 8.

Commit:
Describe which version failed to build

When all images are built at-once, if a particular build fails, it is
now clear what tag failed to build.

Commit:
Update Composer installation method

Directly downloading phars from `getcomposer.org` no longer works for
`wget` or `curl`. The new installation method uses the official setup
for installing composer programmatically.

Commit:
Enable Blackfire

Commit:
Update base file based on php80
